lib/watch-sidebar.js

```diff
@@ -106,13 +105,34 @@ export default class WatchSidebar {
       name: v.getCode(),
       value: k
     }));
-    WatchesPicker.onConfirmed = item => {
-      this.watchViews[item.value].destroy();
-      this.watchViews.splice(item.value, 1);
-      if (this.watchViews.length === 0) this.addWatch();
-    };
-    WatchesPicker.setItems(watches);
-    WatchesPicker.toggle();
+    watchesPicker = new SelectListView({
+      items: watches,
+      elementForItem: watch => {
+        const element = document.createElement("li");
+        element.textContent = watch.name;
+        element.dataset = watch.name;
+        return element;
+      },
+      didConfirmSelection: watch => {
+        this.watchViews[watch.value].destroy();
+        this.watchViews.splice(watch.value, 1);
+        modalPanel.hide();
+        modalPanel.destroy();
+        previouslyFocusedElement.focus();
+      },
+      itemsClassList: ["watch-picker-item"],
+      didCancelSelection: () => {
+        modalPanel.hide();
+        modalPanel.destroy();
+        previouslyFocusedElement.focus();
```

Comment:
`if (previouslyFocusedElement) previouslyFocusedElement.focus();` should do the job.